syntax = "proto3";
package permission;





/*
BatchValidateCmdbPermission请求
*/
message BatchValidateCmdbPermissionRequest {
    /*
    需要鉴权的用户名
    */
    string user = 1 ;
    /*
    需要鉴权的权限点
    */
    string action = 2 ;
    /*
    需要鉴权的object_id
    */
    string object_id = 3 ;
    /*
    需要鉴权的instance_id列表使用逗号分隔
    */
    string instance_id = 4 ;
    /*
    如果一个权限点有数据权限,但是不想校验数据权限,这个参数传true，默认为false, 须传false
    */
    bool validate_action_only = 5 ;
    /*
    是否为批量, 须传true
    */
    bool batch = 6 ;
    /*
    传instanceId,_object_id
    */
    string batch_select_fields = 7 ;
}

/*
BatchValidateCmdbPermission返回
*/
message BatchValidateCmdbPermissionResponse {
message NoPermissionInstances {
    /*
    实例Id
    */
    string instanceId = 1 ;
}
        
    
    /*
    是否鉴权成功
    */
    bool accepted = 1 ;
    /*
    是否为管理员
    */
    bool is_admin = 2 ;
    /*
    有权限的实例
    */
    repeated string authorizers = 3 ;
    /*
    不存在的实例
    */
    repeated string not_found_instances = 4 ;
    /*
    无权限的实例
    */
    repeated NoPermissionInstances no_permission_instances = 5 ;
}

/*
BatchValidateCmdbPermissionApi返回
*/
message BatchValidateCmdbPermissionResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    permission.BatchValidateCmdbPermissionResponse data = 4 ;
}

